core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -414,13 +414,6 @@ private static BulkItemResultHolder executeUpdateRequest(UpdateRequest updateReq
         FAILURE
     }
 
-    static {
-        assert Version.CURRENT.minimumCompatibilityVersion().after(Version.V_5_0_0) == false:
```

Comment:
I removed it and spoke to @bleskes he doens't know what it means and I have a test failure so I am just leaving it as it is.